File: hw/vaca/u3/MazeGenerator/src/main/java/ec/edu/espe/mazegenerator/model/Maze.java
package ec.edu.espe.mazegenerator.model;

/**
 *
 * @author Kevin Vaca Edison's OOP ESPE
 */
public class Maze {
    private final int width, height;
    private final MazeRoom[][] grid;

    public Maze(int width, int height) {
        this.width = width;
        this.height = height;
        grid = new MazeRoom[height][width];
        for (int y = 0; y < height; y++)
            for (int x = 0; x < width; x++)
                grid[y][x] = new MazeRoom();
    }
    
    public MazeRoom getRoom(int x, int y) {
        return grid[y][x];
    }

    public int getWidth() { 
        return width; 
    }
    public int getHeight() { 
        return height; 
    }
}

File: hw/vaca/u3/MazeGeneratorJs/Maze.js
import { MazeRoom } from "./MazeRoom.js";

export class Maze {
  constructor(width, height) {
    this.width = width;
    this.height = height;
    this.rooms = Array.from({ length: height }, () =>
      Array.from({ length: width }, () => new MazeRoom())
    );
  }

  getWidth() {
    return this.width;
  }

  getHeight() {
    return this.height;
  }

  getRoom(x, y) {
    return this.rooms[y][x];
  }
}
